### PR TITLE
Add an autocomplete form input

### PR DESCRIPTION
**Pull request by camertron**
### What are you trying to accomplish?

This PR adds an autocomplete form input to the forms framework.

### Screenshots

|Input with caption and validation message|With dropdown list|
|-|-|
|A screenshot of the autocomplete form input, complete with a label, caption, and validation message.|A screenshot of the autocomplete form input. A single character, the letter 'a', has been typed into the text field. A dropdown list appears showing various fruits that start with that letter.|

### Integration

No changes necessary in production.

#### Risk Assessment
  

- [x] **Low risk** the change is small, highly observable, and easily rolled back.
- [ ] **Medium risk** changes that are isolated, reduced in scope or could impact few users. The change will not impact library availability.
- [ ] **High risk** changes are those that could impact customers and SLOs, low or no test coverage, low observability, or slow to rollback.

### What approach did you choose and why?

I chose to re-use the existing `Primer::Beta::AutoComplete` component.

### Accessibility

- **No new axe scan violation** - This change does not introduce any new [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violations.

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [x] Added/updated previews (Lookbook)
- [x] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

**Comment by camertron** [comment]
Depends on: https://github.com/github/auto-complete-element/pull/116

**Review by langermank — APPROVED** [review]

File: app/components/primer/alpha/dialog.pcss [hunk 1]
```diff
@@ -21,6 +21,15 @@ dialog.Overlay:not([open]) {
...
 
+/* hacks until firefox fully supports popovers */
+:where([popover]:not(.\:popover-open)) {
+  display: none !important;
+}
+
+:popover-open {
+  display: inherit !important;
+}
+
```
Comment (jonrohan) [inline app/components/primer/alpha/dialog.pcss:32]: This seems a bit globally dangerous, will it effect current popovers on the sight?

**Comment by camertron** [comment]
Now depends on: https://github.com/github/auto-complete-element/pull/118

File: app/components/primer/alpha/dialog.pcss (hunk 1 above)
Comment (camertron) [reply app/components/primer/alpha/dialog.pcss:32]: I'm not sure 🤔 If it does, it should only affect Firefox users. It seems to work ok in Lookbook, but some careful testing will need to be done in dotcom. I will integrate and test before merging this PR.

Comment (camertron) [reply app/components/primer/alpha/dialog.pcss:32]: Integration testing PR: https://github.com/github/github/pull/315906

**Review by jonrohan — APPROVED** [review]

**Pull request merged**